File: packages/contracts/src/asset.ts
import { z } from "zod";

export const assetSchema = z.object({
  id: z.string(),
  assetName: z.string(),
  assetType: z.enum(["server", "database", "domain-controller", "bastion", "firewall", "application"]),
  protectionObjectId: z.string(),
  securityDomainId: z.string(),
  businessSystemId: z.string().optional(),
  valueLevel: z.number().min(1).max(5),
  isInternetExposed: z.boolean().default(false),
  managementEntry: z.string().optional()
});

export type Asset = z.infer<typeof assetSchema>;

File: apps/api/app/schemas/asset.py
from pydantic import BaseModel, ConfigDict


class AssetCreate(BaseModel):
    asset_name: str
    asset_type: str
    protection_object_id: int
    security_domain_id: int
    value_level: int


class AssetRead(AssetCreate):
    model_config = ConfigDict(from_attributes=True)

    id: int
